include/psa/crypto_struct.h

```diff
@@ -247,6 +247,25 @@ typedef struct psa_tls12_prf_key_derivation_s {
 #endif /* MBEDTLS_PSA_BUILTIN_ALG_TLS12_PRF) ||
         * MBEDTLS_PSA_BUILTIN_ALG_TLS12_PSK_TO_MS */
 
+#if defined(MBEDTLS_PSA_BUILTIN_ALG_PBKDF2_HMAC)
+typedef enum {
+    PSA_PBKDF2_STATE_INIT,             /* no input provided */
+    PSA_PBKDF2_STATE_INPUT_COST_SET,   /* input cost has been set */
+    PSA_PBKDF2_STATE_SALT_SET,         /* salt has been set */
+    PSA_PBKDF2_STATE_PASSWORD_SET,     /* password has been set */
+    PSA_PBKDF2_STATE_OUTPUT            /* output has been started */
+} psa_pbkdf2_state_t;
+
+typedef struct {
```

Comment:
Related to PR sequencing, so this is not a must for me, but I personally think that we should only introduce members that we are using at this point of the implementation. Related to my latest reply to [this comment](https://github.com/silabs-Kusumit/mbedtls/pull/2#discussion_r1155532162).

Reply:
Isn't it better to add all the struct members in one PR? We are anyways going to add the `pbkdf2_state` enum, in which I think, all the states should be added in one go as they reference the other input steps.